### PR TITLE
Initial commit for Apache Cassandra support in OpenEBS.

### PR DESCRIPTION
**Pull request by yudaykiran**
**What this PR does / why we need it**:
- Creates Headless Service for Apache Cassandra.
- Creates StatefulSet for Apache Cassandra with ReplicaSet set to 3.
- Uses OpenEBS Storage in volumeClaimTemplates.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #
Fixes issue #458

**Special notes for your reviewer**:

Signed-off-by: yudaykiran <udaykiran.y@gmail.com>

### CODE REVIEW

**Comment by yudaykiran** [comment]
Creating Cassandra Headless Service:

```
ubuntu@o-master01:~$ kubectl apply -f cassandra-service.yaml 
service "cassandra" created
```

Creating Cassandra StatefulSet:

```
ubuntu@o-master01:~$ kubectl apply -f cassandra-statefulset.yaml 
statefulset "cassandra" created
```

Output of StatefulSets:

```
ubuntu@o-master01:~$ kubectl get statefulset
NAME        DESIRED   CURRENT   AGE
cassandra   3         3         4m
```

Output of Services:

```
ubuntu@o-master01:~$ kubectl get svc
NAME                                                CLUSTER-IP       EXTERNAL-IP   PORT(S)             AGE
cassandra                                           None             <none>        9042/TCP            7m
kubernetes                                          10.96.0.1        <none>        443/TCP             8d
maya-apiserver-service                              10.96.59.13      <none>        5656/TCP            7d
pvc-2137fb74-a98a-11e7-9784-000c29119159-ctrl-svc   10.100.200.96    <none>        3260/TCP,9501/TCP   7m
pvc-5beacf17-a98a-11e7-9784-000c29119159-ctrl-svc   10.101.234.102   <none>        3260/TCP,9501/TCP   5m
pvc-962c2e7e-a98a-11e7-9784-000c29119159-ctrl-svc   10.96.147.48     <none>        3260/TCP,9501/TCP   4m
```

Output of Pods after creation of Apache Cassandra StatefulSets:

```
ubuntu@o-master01:~$ kubectl get pods -o wide --all-namespaces
NAMESPACE     NAME                                                             READY     STATUS    RESTARTS   AGE       IP             NODE
default       cassandra-0                                                      1/1       Running   0          6m        10.44.0.3      o-host01
default       cassandra-1                                                      1/1       Running   0          5m        10.36.0.6      o-host02
default       cassandra-2                                                      1/1       Running   0          3m        10.44.0.7      o-host01
default       maya-apiserver-546443015-0zbhc                                   1/1       Running   3          7d        10.36.0.1      o-host02
default       openebs-provisioner-3862123968-rhdxq                             1/1       Running   3          1d        10.36.0.2      o-host02
default       pvc-2137fb74-a98a-11e7-9784-000c29119159-ctrl-1110762493-85s8k   1/1       Running   0          6m        10.44.0.1      o-host01
default       pvc-2137fb74-a98a-11e7-9784-000c29119159-rep-4006930475-4s5rb    1/1       Running   0          6m        10.36.0.3      o-host02
default       pvc-2137fb74-a98a-11e7-9784-000c29119159-rep-4006930475-gh8r5    1/1       Running   0          6m        10.44.0.2      o-host01
default       pvc-5beacf17-a98a-11e7-9784-000c29119159-ctrl-3554066367-pkb0k   1/1       Running   0          5m        10.36.0.5      o-host02
default       pvc-5beacf17-a98a-11e7-9784-000c29119159-rep-1057563099-v6j1h    1/1       Running   0          5m        10.44.0.4      o-host01
default       pvc-5beacf17-a98a-11e7-9784-000c29119159-rep-1057563099-x0b9q    1/1       Running   0          5m        10.36.0.4      o-host02
default       pvc-962c2e7e-a98a-11e7-9784-000c29119159-ctrl-1152450047-7psnn   1/1       Running   0          3m        10.44.0.6      o-host01
default       pvc-962c2e7e-a98a-11e7-9784-000c29119159-rep-230365072-8p909     1/1       Running   0          3m        10.44.0.5      o-host01
default       pvc-962c2e7e-a98a-11e7-9784-000c29119159-rep-230365072-s88l7     1/1       Running   0          3m        10.36.0.7      o-host02
kube-system   etcd-o-master01                                                  1/1       Running   3          8d        20.10.29.201   o-master01
kube-system   kube-apiserver-o-master01                                        1/1       Running   3          8d        20.10.29.201   o-master01
kube-system   kube-controller-manager-o-master01                               1/1       Running   3          8d        20.10.29.201   o-master01
kube-system   kube-dns-2425271678-zvhcl                                        3/3       Running   15         8d        10.32.0.4      o-master01
kube-system   kube-proxy-4mqrw                                                 1/1       Running   3          8d        20.10.29.201   o-master01
kube-system   kube-proxy-j8hn8                                                 1/1       Running   4          8d        20.10.29.203   o-host02
kube-system   kube-proxy-p3j5b                                                 1/1       Running   4          8d        20.10.29.202   o-host01
kube-system   kube-scheduler-o-master01                                        1/1       Running   3          8d        20.10.29.201   o-master01
kube-system   weave-net-gb7pz                                                  2/2       Running   8          8d        20.10.29.203   o-host02
kube-system   weave-net-gtnsr                                                  2/2       Running   6          8d        20.10.29.201   o-master01
kube-system   weave-net-h0zg6                                                  2/2       Running   8          8d        20.10.29.202   o-host01
```

File: k8s/demo/cassandra/cassandra-statefulset.yaml [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+          timeoutSeconds: 5
+        # These volume mounts are persistent. They are like inline claims,
+        # but not exactly because the names need to match exactly one of
+        # the stateful pod volumes.
+        volumeMounts:
+        - name: cassandra-data
+          mountPath: /cassandra_data
+  # These are converted to volume claims by the controller
+  # and mounted at the paths mentioned above.
+  # do not use these in production until ssd GCEPersistentDisk or other ssd pd
```
Comment (kmova) [inline k8s/demo/cassandra/cassandra-statefulset.yaml]: what does this above comment mean?

Comment (yudaykiran) [reply k8s/demo/cassandra/cassandra-statefulset.yaml]: I picked up the example from kubernetes which was using GCE persistent disk, that comment is not relevant for OpenEBS and needs to be removed.

File: k8s/demo/cassandra/cassandra-statefulset.yaml [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  volumeClaimTemplates:
+  - metadata:
+      name: cassandra-data
+      annotations:
+        volume.beta.kubernetes.io/storage-class: openebs-basic
```
Comment (kmova) [inline k8s/demo/cassandra/cassandra-statefulset.yaml]: Can you also make a change to the openebs-storageclasses.yaml, probably to have a new class called : openebs-cassandra

Comment (yudaykiran) [reply k8s/demo/cassandra/cassandra-statefulset.yaml]: Made the required changes for MongoDB and Apache Cassandra.

**Comment by yudaykiran** [comment]
Changes to OpenEBS Storage Classes:

```
ubuntu@o-master01:~$ kubectl get sc
NAME                TYPE
openebs-basic       openebs.io/provisioner-iscsi   
openebs-cassandra   openebs.io/provisioner-iscsi   
openebs-jupyter     openebs.io/provisioner-iscsi   
openebs-mongodb     openebs.io/provisioner-iscsi   
openebs-percona     openebs.io/provisioner-iscsi
```

MongoDB StatefulSet using openebs-mongodb storage class:

```
ubuntu@o-master01:~$ kubectl get pvc
NAME                               STATUS    VOLUME                                     CAPACITY   ACCESSMODES   STORAGECLASS      AGE
mongo-persistent-storage-mongo-0   Bound     pvc-e3eef8da-aa60-11e7-9784-000c29119159   5G         RWO           openebs-mongodb   1h
mongo-persistent-storage-mongo-1   Bound     pvc-1e3ebf6f-aa61-11e7-9784-000c29119159   5G         RWO           openebs-mongodb   1h
mongo-persistent-storage-mongo-2   Bound     pvc-2d78cee0-aa61-11e7-9784-000c29119159   5G         RWO           openebs-mongodb   1h

```

Apache Cassandra using openebs-cassandra stroage class:

```
ubuntu@o-master01:~$ kubectl get pvc
NAME                         STATUS    VOLUME                                     CAPACITY   ACCESSMODES   STORAGECLASS        AGE
cassandra-data-cassandra-0   Bound     pvc-f015ba66-aa6b-11e7-9784-000c29119159   5G         RWO           openebs-cassandra   5m
cassandra-data-cassandra-1   Bound     pvc-1fb335d7-aa6c-11e7-9784-000c29119159   5G         RWO           openebs-cassandra   4m
cassandra-data-cassandra-2   Bound     pvc-49b48954-aa6c-11e7-9784-000c29119159   5G         RWO           openebs-cassandra   3m
```

File: k8s/demo/cassandra/cassandra-statefulset.yaml (hunk 2 above)
Comment (yudaykiran) [reply k8s/demo/cassandra/cassandra-statefulset.yaml]: Done

**Pull request merged**